api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -309,6 +309,18 @@ public static <T> UnboundTerm<T> transform(String name, Transform<?, T> transfor
     return new UnboundTransform<>(ref(name), transform);
   }
 
+  /**
+   * Create a {@link Literal} from an Object.
+   *
+   * @param value a value
+   * @param <T> Java type of value
+   * @return a Literal for the given value
+   * @throws IllegalArgumentException if the value has no literal implementation
+   */
+  public static <T> Literal<T> lit(T value) {
```

Comment:
I am not quite clear on the need for literals. Literals are part of expressions but not sure if default values are first-class expressions. I see we could use them in expressions, but as part of the `NestedField` API, I do not see a need to bring in expressions yet.

Also note that the input value comes from `SingleValueParser` which returns Iceberg data. I am guessing the input to `Literals` is weakly defined now (since it is constrained to primitives) and it is not clear if it will be Iceberg data or Java data, etc?

Reply:
The purpose of this is to support the same set of values as expressions do.

Adding the default value methods to the API creates a weird situation in which we need to coerce values from `Object` to a value of the appropriate type and internal representation. Rather than create completely different code to do that, I'm reusing what we already support for expressions by creating a literal (which handles coercing from `Object`) and then converting it to a specific type using `to` (which handles conversion to the right Iceberg representation).

I think it's important to have consistency here.